File: buerguer-pig/HamburgesasEIngredientes-Java-IvanRC/src/main/java/factory/IngredienteFactory.java
package factory;

import models.Ingrediente;

import java.util.ArrayList;

public class IngredienteFactory {
    private IngredienteFactory() {}
    private static IngredienteFactory instance = null;

    public static IngredienteFactory getInstance() {
        if(instance == null) instance = new IngredienteFactory();
        return instance;
    }

    private static int contador = 0;

    private static int nextId(){
        contador++;
        return contador;
    }

    public ArrayList<Ingrediente> getSome(){
        ArrayList<Ingrediente> ingredientes = new ArrayList<Ingrediente>();
        int number = (int)(Math.random()*5+1);
        String[] nombres = {"Salsa", "Carne", "Lechuga", "Cebolla", "Pepinillo", "Pan"};
        for(int i = 1; i<=number; i++){
            double precio = (Math.random()*100.0);
            ingredientes.add(
                    new Ingrediente(nextId(), nombres[(int)(Math.random()* nombres.length)], precio)
            );
        }
        return ingredientes;
    }
}

File: buerguer-pig/HamburgesasEIngredientes-Kotlin-IvanRC/src/main/kotlin/factory/IngredienteFactory.kt
package factory

import model.Ingrediente
import mu.KotlinLogging

class IngredienteFactory {

    private val logger = KotlinLogging.logger {}

    companion object {
        var contador = 1
        fun getNextId(): Int{
            return contador++
        }

        private var instance: IngredienteFactory? = null

        fun getInstance(): IngredienteFactory{
            if(instance == null){
                instance = IngredienteFactory()
            }
            return instance!!
        }
    }

    fun createSomeRandom(): List<Ingrediente> {
        logger.debug { "Se inicia el factory que creará algún ingrediente aleatorio" }
        val ingredientes = mutableListOf<Ingrediente>()
        val nombres = arrayOf("Salsa", "Carne", "Lechuga", "Cebolla", "Pepinillo", "Pan")
        val numero = (1..6).random()
        for(i in 1..numero){
            val precio: Double = ((0..750).random() / 100.0)
            var ingrediente: Ingrediente = Ingrediente(getNextId(), nombres.random(), precio)
            ingredientes.add(ingrediente)
        }
        return ingredientes.toList()
    }

}
